data/src/main/java/com/linkedin/data/schema/PdlBuilder.java

```diff
@@ -275,11 +274,11 @@ private String toJson(Object value) throws IOException
   {
     if (value instanceof DataMap)
     {
-      return JSON_CODEC.mapToString((DataMap) value);
+      return _jsonCodec.mapToString((DataMap) value);
```

Comment:
Cleaner option is to create two protected abstrac methods toJson(DataMap) toJson(DataList) and let concrete implementations decide how the json is generated.

Reply:
why not have only 1 ie toJson(Object) and writeJson is kind of fulfilling that but if needed we can separate out toJson